tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/op/NameScope.java

```diff
@@ -50,6 +53,45 @@ NameScope withName(String name) {
     return new NameScope(opPrefix, name, ids);
   }
 
+  private static final Pattern NAME_PATTERN = Pattern.compile("(.+)_(\\d+)", Pattern.DOTALL);
+
+  /**
+   * "Import" used names from a graph.  Useful when adding to a loaded graph.
+   */
+  NameScope withUsedFrom(ExecutionEnvironment env) {
```

Comment:
Why does this accept an `ExecutionEnvironment` rather than a `Graph` if it only works on graphs?

Reply:
For ease of calling it from Scope.  Scope and Ops work with ExecutionEnvironments, and it's easier to check it here than there.